### ISSUE TITLE
Application user data size grows when application kill by task manager

### ISSUE THREAD

**Issue by expcapitaldev** [smartlook/smartlook-mobile-issue-tracker#69]
**Platform**

Android  7/8/9/10/11 (version does not matter, the bug is reproducible on several versions)

**Technology**

Cordova 9.0.0
Ionic 5.2.2

**Build instructions [optional]**

----------------gradle dependencies ---------------------
```
       buildscript {
            ext.kotlin_version = '1.3.50'
            repositories {
                google()
                jcenter()
            }
            dependencies {
                classpath 'com.android.tools.build:gradle:4.0.0'
                classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
            }
        }

      allprojects {
          repositories {
              google()
              jcenter()
          }
      
          project.ext {
            defaultBuildToolsVersion="29.0.2" //String
            defaultMinSdkVersion=22 
            defaultTargetSdkVersion=29
            defaultCompileSdkVersion=29
          }
      }

      task clean(type: Delete) {
          delete rootProject.buildDir
      }
```

----------------------------------------------------------------------

cordova plugin list 

cordova-android-support-gradle-release 3.0.1 "cordova-android-support-gradle-release"
cordova-clipboard 1.3.0 "Clipboard"
cordova-custom-config 5.1.0 "cordova-custom-config"
cordova-launch-review 3.1.0 "Launch Review"
cordova-plugin-add-swift-support 2.0.2 "AddSwiftSupport"
cordova-plugin-android-idfa 1.0.4 "Advertising ID (IDFA) Plugin using Google Play Services for Android"
cordova-plugin-androidx 3.0.0 "cordova-plugin-androidx"
cordova-plugin-androidx-adapter 1.1.3 "cordova-plugin-androidx-adapter"
cordova-plugin-camera 4.1.0 "Camera"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-file 6.0.1 "File"
cordova-plugin-globalization 1.0.7 "Globalization"
cordova-plugin-inappbrowser-updated 4.1.0-dev "InAppBrowser"
cordova-plugin-ionic-keyboard 2.1.2 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 4.2.1 "cordova-plugin-ionic-webview"
cordova-plugin-keychain-touch-id 3.3.0 "TouchID and Keychain"
cordova-plugin-nativestorage 2.2.2 "NativeStorage"
cordova-plugin-network-information 2.0.1 "Network Information"
cordova-plugin-screen-orientation 2.0.2 "Screen Orientation"
cordova-plugin-smartlook 1.7.4 "Smartlook"
cordova-plugin-splashscreen 6.0.0 "Splashscreen"
cordova-plugin-statusbar 2.4.4-dev "StatusBar"
cordova-plugin-whitelist 1.3.2 "Whitelist"
cordova-universal-links-plugin 1.2.1 "Universal Links Plugin"
cordova.plugins.diagnostic 4.0.12 "Diagnostic"
es6-promise-plugin 4.2.2 "Promise"

**Rendering [optional]**

**Affected devices [optional]**

**Smartlook SDK version**

1.7.3 (with 1.7.4 also reproduced)

**Summary**

When you enter the application and close it through the task manager, the application memory starts to grow. In particular, from 10MB to 50MB is added per session. The amount of added cache is also affected by the number of DOM nodes in the presentation template as well as the CSS that is used in this template.

**Steps to reproduce [optional]**

**Example Project [optional]**

**Relevant logs and/or screenshots**

https://user-images.githubusercontent.com/19390317/109814235-52de0d00-7c3f-11eb-9e0c-781a0b0ed45c.mp4

**Comment by SenNeonoveNoci** [comment]
Hello @expcapitaldev this an expected behaviour. If you close the application we are not able to upload the record data, so we need to store them locally so they can be uploaded on the next application run.

After a successful upload, the data is erased.

This cache has some limitations:
- The maximum cache size is **200MB** or if a device has low free memory as a 20% of a left space (must be smaller than 200MB)
- In case there is less than **50MB** of free space on a device, the cache is not used at all

**Comment by expcapitaldev** [comment]
![image](https://user-images.githubusercontent.com/19390317/109821246-d6e7c300-7c46-11eb-8eaf-a9046a817c97.png)
![image](https://user-images.githubusercontent.com/19390317/109821258-d9e2b380-7c46-11eb-91a7-85e067a96931.png)
By such actions, as shown in the video, you can fill the phone's memory all the way and the size of the application is limited in this case only by the amount of memory in the phone itself

**Comment by SenNeonoveNoci** [comment]
@expcapitaldev Ok this is not expected behaviour indeed. I'm gonna test this and keep you updated. Sorry for the inconvenience.

**Comment by AndrewMorsillo** [comment]
@SenNeonoveNoci We may also be seeing this on the react native SDK. We've had users complaining that our app is filling up their phone's storage over time despite our app not storing any significant amount of data. It sounds like this issue.

**Comment by AndrewMorsillo** [comment]
@SenNeonoveNoci I have confirmed this bug in my react native app on Android. When Smartlook (smartlook-react-native-wrapper 0.4.0) is installed the storage of the app grows by 5-15mb every time the app is launched and subsequently closed. 

A fresh install of my app uses ~60mb of storage. After starting and closing the app a number of times this ballooned to over 500mb and continued to grow. I made an identical build excluding Smartlook and the issue was no longer present.

This is an extremely serious bug. If we can't find a resolution quickly we will be forced to remove Smartlook from our app.

**Comment by krystofcelba** [comment]
Hey, we've recently released a new version `1.2.1` and `1.9.1` for React Native and Cordova, respectively. Among other things we've also addressed the issues, you reported. Please do the update and in case of any issue let us know. Thanks!
